### PR TITLE
Fix mediaPlayer finalized without being released

### LINKED ISSUE #38: MediaPlayer finalized without being released

### PR DESCRIPTION
**Pull request by ghsdh3409**
* It closes #38.
* It resolves https://github.com/hayanmind/mimiking-rn/issues/1663.

### CODE REVIEW

**Comment by ghsdh3409** [comment]
@ohtangza This is ready to review!

File: android/src/main/java/com/zmxv/RNSound/RNSoundModule.java [hunk 1]
```diff
@@ -145,6 +144,9 @@ public synchronized void onPrepared(MediaPlayer mp) {
 
       @Override
       public synchronized boolean onError(MediaPlayer mp, int what, int extra) {
+        if (module.playerPool.get(key) != null) {
```
Comment (ohtangza) [inline android/src/main/java/com/zmxv/RNSound/RNSoundModule.java:147]: So, I'm wondering if the `key` variable safe even after return.

**Review by ohtangza — APPROVED** [review]
Looks very clear! I commented a little question (not critical).

File: android/src/main/java/com/zmxv/RNSound/RNSoundModule.java (hunk 1 above)
Comment (ghsdh3409) [reply android/src/main/java/com/zmxv/RNSound/RNSoundModule.java:147]: Wow... I am not sure, but I guess, `key` was set to `playerPool`. Thus, it maybe safe...?
I will check later!

**Pull request merged**